lib/services.js

```diff
@@ -27,41 +27,50 @@ async function startWebInspectorService (udid, opts = {}) {
   if (!semverVersion) {
     throw new Error(`Could not create a semver version out of '${osVersion}'`);
   }
-  const socketClient = opts.socket || await startService(udid, WEB_INSPECTOR_SERVICE_NAME);
-  return new WebInspectorService({
+  if (opts.socket) {
+    return new WebInspectorService({
+      majorOsVersion: semverVersion.major,
+      isSimulator,
+      socketChunkSize,
+      verbose,
+      verboseHexDump,
+      socketClient: opts.socket,
+    });
+  }
+  return await startService(udid, WEB_INSPECTOR_SERVICE_NAME, undefined, socketClient => new WebInspectorService({
     majorOsVersion: semverVersion.major,
     isSimulator,
     socketChunkSize,
     verbose,
     verboseHexDump,
     socketClient,
-  });
+  }));
 }
 
 async function startInstallationProxyService (udid, opts = {}) {
-  return new InstallationProxyService(new PlistService(await startService(udid, INSTALLATION_PROXY_SERVICE_NAME, opts.socket)));
+  return await startService(udid, INSTALLATION_PROXY_SERVICE_NAME, opts.socket, socket => new InstallationProxyService(new PlistService(socket)));
```

Comment:
We aurally put lama args into parentheses for better reading. Also consider splitting long lines

Reply:
llama args? :)

Reply:
sorry, lambda of course